### PR TITLE
DHFPROD-8567: Recent Searches widget

### PR DESCRIPTION
**Pull request by wooldridge**
RecentSearches writes to and pulls from local browser storage.
Name widget components RecentSearches and RecentRecords and refactor variable naming.
Add new tests for RecentSearches.
Update broken tests for other components due to renaming in context.
Add font libraries to package.json.
Style tweaks across app.

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests  

- ##### Reviewer:

- [ ] Reviewed Tests
- [ ] Added to Release Wiki

### CODE REVIEW

File: marklogic-data-hub-central/ui-custom/src/views/Detail.tsx [hunk 1]
```diff
@@ -115,7 +115,7 @@ const Detail: React.FC<Props> = (props) => {
                     "backgroundColor": "transparent"
                   },
                   "mainStyle": {
-                    "padding-top": "8px"
+                    "paddingTop": "8px"
```
Comment (aniamy) [inline marklogic-data-hub-central/ui-custom/src/views/Detail.tsx:118]: Careful here. This PR and Ivey's PR have the same change. Shouldn't it be padding-top?

File: marklogic-data-hub-central/ui-custom/src/components/RecentSearches/RecentSearches.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        <span className="icon" onClick={handleShareClick(row)} data-testid="share-icon">
+          <FontAwesomeIcon size={"lg"} icon={faShareSquare} style={{color: "#394494"}}></FontAwesomeIcon>
+        </span>
+      </OverlayTrigger>;
+    }
+  }
+
+  return (
+    <div className="recentSearches">
+      {(props.data && props.data.length > 0) ? (
```
Comment (aniamy) [inline marklogic-data-hub-central/ui-custom/src/components/RecentSearches/RecentSearches.tsx:80]: Rememeber to destructure props.
https://wiki.marklogic.com/display/ENGINEERING/Data+Hub+Central+React+Best+Practices

File: marklogic-data-hub-central/ui-custom/src/views/Detail.tsx (hunk 1 above)
Comment (wooldridge) [reply marklogic-data-hub-central/ui-custom/src/views/Detail.tsx:118]: Thanks for reviewing, @aniamy. React components take JS objects when applying CSS rules inline, so the CSS rules need to be written in camelCase. paddingTop, backgroundColor, etc...

File: marklogic-data-hub-central/ui-custom/src/components/RecentSearches/RecentSearches.tsx (hunk 2 above)
Comment (wooldridge) [reply marklogic-data-hub-central/ui-custom/src/components/RecentSearches/RecentSearches.tsx:80]: @aniamy, thanks, yes, I need to refactor.

Comment (wooldridge) [reply marklogic-data-hub-central/ui-custom/src/components/RecentSearches/RecentSearches.tsx:80]: @aniamy we will do some of the pattern refactoring across the components soon. We need to merge this now for the meeting today.

**Pull request merged**